src/idl_parser.cpp

```diff
@@ -1649,6 +1673,16 @@ CheckedError Parser::ParseEnum(bool is_union, EnumDef **dest) {
       if (Is('=')) {
         NEXT();
         ECHECK(atot(attribute_.c_str(), *this, &ev.value));
+        ev.value = StringToInt(attribute_.c_str());
```

Comment:
Yes, you are right.  Using `ParseSingleValue` makes the code much shorter now.  I've kept denying boolean enums even though you could define `enum X:bool { A = true, B = false }` which could make sense because it doesn't currently forbid `enum X:bool { A = 100 }` which doesn't make sense in my opinion.  It would also need special handling in the code generators, I guess.

Reply:
If @aardappel agree with bool-base enums it will be very simple:
- comment or remove one line of code
- update grammar documentation

 Boolean values as a base for enum was disabled before refactoring, I only save this behavior.

Reply:
I don't see a use for boolean enums, since `bool` in FlatBuffers always takes one byte. So you get the same as using `ubyte` except now your enum is not extensible. I'd suggest to not allow `bool`.

Reply:
@aardappel 
Is following declaration acceptable?
`enum X:ubyte { A = true, B = false }`

Or simply: Can boolean constants **true** and **false** are used for implicit initialization (1, 0) of an integer-type values?
Now **true** or **false** can be applied only to a bool-type value.

Reply:
I suppose we could allow these as integers in any context (by moving them to be parsed as integers in the lexical analysis), though I don't see the need / use case for this.